### PR TITLE
PoolformerImageProcessor defaults to match previous FE

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Output of PoolformerImageProcessor didn't exactly match previous feature extractor. Updates defaults and size calculation logic to match outputs. 

Running:
```
import torch
from transformers import PoolFormerFeatureExtractor, PoolFormerModel
from datasets import load_dataset
dataset = load_dataset("huggingface/cats-image")
image = dataset["test"]["image"][0]

feature_extractor = PoolFormerFeatureExtractor.from_pretrained("sail/poolformer_s12")
model = PoolFormerModel.from_pretrained("sail/poolformer_s12")

inputs = feature_extractor(image, return_tensors="pt")

with torch.no_grad():
    outputs = model(**inputs)

last_hidden_states = outputs.last_hidden_state
r = list(last_hidden_states.shape)
print(r)
```

Now outputs images of size `[1, 512, 7, 7]` - matching the output of the images before #19796 was merged in. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/poolformer/image_processing_poolformer.py [hunk 1]
```diff
@@ -184,10 +184,7 @@ def resize(
                 if size["height"] == size["width"]:
                     scale_size = int(math.floor(size["height"] / crop_pct))
                 else:
-                    scale_size = (
-                        int(math.floor(size["height"] / crop_pct)),
-                        int(math.floor(size["width"] / crop_pct)),
-                    )
+                    scale_size = (int(size["height"] / crop_pct), int(size["width"] / crop_pct))
```
Comment (ydshieh) [inline src/transformers/models/poolformer/image_processing_poolformer.py:187]: I am wondering if this change is necessary to have the test pass. I see this is what has been in the previous version, but don't think this matter.

In fact, `math.floor` in this file is not necessary (at least for Python 3.x). Anyway, using it or not are both fine to me, but keeping the same style would be good (as long as my claim is correct - i.e. it doesn't affect the test we are trying to fix)

**Review by ydshieh — APPROVED** [review]
Thank you, @amyeroberts . Left a comment regarding the style.

File: src/transformers/models/poolformer/image_processing_poolformer.py (hunk 1 above)
Comment (amyeroberts) [reply src/transformers/models/poolformer/image_processing_poolformer.py:187]: You're completely right. Tbh, I was just porting from the old feature extractors without thinking too much. I'll remove the `math.floor` in the function to make it tidier 🧹

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

**Pull request merged**